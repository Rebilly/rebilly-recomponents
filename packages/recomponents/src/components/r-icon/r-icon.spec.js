import {shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RIcon from './r-icon.vue';

describe('r-icon.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RIcon, {
            propsData: {icon: 'user'},
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = await renderToString(RIcon, {
            propsData: {icon: 'user'},
        });

        expect(wrapper).toMatchSnapshot();
    });
});
