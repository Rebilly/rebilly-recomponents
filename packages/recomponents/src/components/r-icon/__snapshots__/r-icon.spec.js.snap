// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-icon.vue should render Wrapper and match snapshot 1`] = `
<svg class="r-icon r-icon-20">
  <use xlink:href="#icon-user" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
</svg>
`;

exports[`r-icon.vue should render via SSR and match snapshot 1`] = `
<svg class="r-icon r-icon-20">
  <use xlink:href="#icon-user" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
</svg>
`;
